### PR TITLE
refactor: add new properties to app definitions

### PR DESCRIPTION
**Pull request by nliu-ms**
1) Add new properties to app definition interfaces
2) Fix api spec file path
3) Show codelens only for copilot plugin project

### CODE REVIEW

File: packages/fx-core/src/core/FxCoreImplementV3.ts [hunk 1]
```diff
@@ -663,15 +669,27 @@ export class FxCoreV3Implement {
...
+      isYaml ? apiSpecYamlFileName : apiSpecJsonFileName
+    );
     const manifestPath = inputs[QuestionNames.ManifestPath];
     const specParser = new SpecParser(openapiSpecPath);
     const adaptiveCardFolder = path.join(
       inputs.projectPath!,
       AppPackageFolderName,
       AdaptiveFolderName
     );
+    await fs.ensureDir(adaptiveCardFolder);
```
Comment (yuqizhou77) [inline packages/fx-core/src/core/FxCoreImplementV3.ts:692]: wondering if we need this. I think adaptiveCardFolder will be created in the lib if it does not exist.

File: packages/fx-core/src/core/FxCoreImplementV3.ts [hunk 2]
```diff
@@ -663,15 +669,27 @@ export class FxCoreV3Implement {
...
+    const url = inputs[QuestionNames.ApiSpecLocation] ?? inputs.openAIPluginManifest?.api.url;
+    const apiSpecFolderPath = path.join(
+      inputs.projectPath!,
+      AppPackageFolderName,
+      apiSpecFolderName
+    );
+    await fs.ensureDir(apiSpecFolderPath);
+
+    const isYaml = await isYamlSpecFile(url);
+    const openapiSpecPath = path.join(
```
Comment (yuqizhou77) [inline packages/fx-core/src/core/FxCoreImplementV3.ts:681]: thought a more reliable way is getting the info from manifest?

Comment (yuqizhou77) [reply packages/fx-core/src/core/FxCoreImplementV3.ts:681]: We can get the path from composeExtension.apiSpecFile

File: packages/fx-core/src/core/FxCoreImplementV3.ts [hunk 3]
```diff
@@ -663,18 +663,26 @@ export class FxCoreV3Implement {
...
       AppPackageFolderName,
       AdaptiveFolderName
     );
 
     try {
-      await specParser.generate(manifestPath, operations, openapiSpecPath, adaptiveCardFolder);
+      const specParser = new SpecParser(url);
+      await specParser.generate(manifestPath, operations, outputAPISpecPath, adaptiveCardFolder);
     } catch (e) {
       const error = assembleError(e);
```
Comment (yuqizhou77) [inline packages/fx-core/src/core/FxCoreImplementV3.ts:687]: maybe consider to improve later to handle specParserError?

File: packages/fx-core/src/component/generator/copilotPlugin/generator.ts [hunk 4]
```diff
@@ -45,10 +45,9 @@ import { SpecParserError } from "../../../common/spec-parser/specParserError";
 
 const componentName = "simplified-message-extension-existing-api";
 const templateName = "simplified-message-extension-existing-api";
-const manifestFileName = ManifestTemplateFileName;
-const apiSpecFolderName = "apiSpecFiles";
-const apiSpecYamlFileName = "openapi.yaml";
-const apiSpecJsonFileName = "openapi.json";
+export const apiSpecFolderName = "apiSpecFiles";
```
Comment (yuqizhou77) [inline packages/fx-core/src/component/generator/copilotPlugin/generator.ts:48]: no need to export?

**Review by yuqizhou77 — APPROVED** [review]

**Review by yuqizhou77 — APPROVED** [review]

**Review by yuqizhou77 — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Pull request merged**